### PR TITLE
Zenoh-C z_slice and z_bytes types compatibility

### PR DESCRIPTION
**Pull request by jean-roland**
This PR will solve #414.

To-do list:
- [x] Rename current bytes types as slice types
- [x] Replace direct slice access with methods `z_slice_len` and `z_slice_data`
- [x] (Re)Introduce bytes type with private internals
- [x] Introduce methods to encode/decode bytes to and from regular types

### CODE REVIEW

**Comment by jean-roland** [comment]
Could you review it @sashacmc

**Review by sashacmc — CHANGES_REQUESTED** [review]

File: include/zenoh-pico/net/encoding.h [hunk 1]
```diff
@@ -15,13 +15,13 @@
...
 #include "zenoh-pico/api/constants.h"
-#include "zenoh-pico/collections/bytes.h"
+#include "zenoh-pico/collections/slice.h"
 
 /**
  * A zenoh encoding.
  */
 typedef struct _z_encoding_t {
-    _z_bytes_t schema;
+    _z_slice_t schema;
```
[the remark this answers is not in the document]
Comment (sashacmc) [reply include/zenoh-pico/net/encoding.h:24]: I created issue for it
https://github.com/eclipse-zenoh/zenoh-pico/issues/418

File: examples/arduino/z_sub.ino [hunk 2]
```diff
@@ -37,7 +37,8 @@
 void data_handler(const z_loaned_sample_t *sample, void *arg) {
     z_owned_string_t keystr;
     z_keyexpr_to_string(z_sample_keyexpr(sample), &keystr);
-    std::string val((const char *)z_sample_payload(sample)->start, z_sample_payload(sample)->len);
+    const z_loaned_bytes_t *payload = z_sample_payload(sample);
+    std::string val((const char *)z_bytes_data(payload), z_bytes_len(payload));
```
[the remark this answers is not in the document]
Comment (jean-roland) [reply examples/arduino/z_sub.ino:41]: I see

File: include/zenoh-pico/collections/slice.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+#include <stddef.h>
+#include <stdint.h>
+
+/*-------- Slice --------*/
+/**
+ * An array of bytes.
+ *
+ * Members:
+ *   size_t len: The length of the bytes array.
+ *   uint8_t *val: A pointer to the bytes array.
```
Comment (sashacmc) [inline include/zenoh-pico/collections/slice.h:28]:
```suggestion
 *   uint8_t *start: A pointer to the bytes array.
```

**Review by sashacmc — APPROVED** [review]

File: src/collections/slice.c [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+uint32_t _z_bytes_to_uint32(const _z_bytes_t *bs) {
+    uint32_t val = 0;
+    memcpy(&val, bs->_slice.start, sizeof(val));
+    return val;
+}
+
+uint64_t _z_bytes_to_uint64(const _z_bytes_t *bs) {
+    uint64_t val = 0;
+    memcpy(&val, bs->_slice.start, sizeof(val));
```
[the remark this answers is not in the document]
Comment (milyin) [reply src/collections/slice.c:179]: To be fixed separately, issue #420

**Review by milyin — APPROVED** [review]

**Pull request merged**